File: src/phorge/api/endpoints/ssh_keys.py
"""SSH key API endpoints."""

from __future__ import annotations

from phorge.api.client import ForgeClient
from phorge.api.models import SSHKey


class SSHKeysAPI:
    def __init__(self, client: ForgeClient) -> None:
        self._client = client

    async def list(self, server_id: int) -> list[SSHKey]:
        data = await self._client.get(f"/servers/{server_id}/keys")
        return [SSHKey(**k) for k in data.get("keys", [])]

    async def get(self, server_id: int, key_id: int) -> SSHKey:
        data = await self._client.get(f"/servers/{server_id}/keys/{key_id}")
        return SSHKey(**data["key"])

    async def create(self, server_id: int, name: str, key: str, username: str = "forge") -> SSHKey:
        data = await self._client.post(
            f"/servers/{server_id}/keys",
            json={"name": name, "key": key, "username": username},
        )
        return SSHKey(**data["key"])

    async def delete(self, server_id: int, key_id: int) -> None:
        await self._client.delete(f"/servers/{server_id}/keys/{key_id}")

File: internal/forge/ssh_keys.go
package forge

import (
	"context"
	"net/http"
)

// List returns all SSH keys on a server.
func (s *SSHKeysService) List(ctx context.Context, serverID int64) ([]SSHKey, error) {
	return listResources[SSHKey](s.client, ctx, s.client.orgPath("/servers/%d/ssh-keys", serverID))
}

// Get returns a single SSH key by ID.
func (s *SSHKeysService) Get(ctx context.Context, serverID, keyID int64) (*SSHKey, error) {
	return getResource[SSHKey](s.client, ctx, s.client.orgPath("/servers/%d/ssh-keys/%d", serverID, keyID))
}

// Create installs a new SSH key on a server.
func (s *SSHKeysService) Create(ctx context.Context, serverID int64, name, key, username string) (*SSHKey, error) {
	body := map[string]string{
		"name":     name,
		"key":      key,
		"username": username,
	}
	return createResource[SSHKey](s.client, ctx, s.client.orgPath("/servers/%d/ssh-keys", serverID), body)
}

// Delete removes an SSH key from a server.
func (s *SSHKeysService) Delete(ctx context.Context, serverID, keyID int64) error {
	return s.client.do(ctx, http.MethodDelete, s.client.orgPath("/servers/%d/ssh-keys/%d", serverID, keyID), nil, nil)
}
